include/openssl/core_names.h

```diff
@@ -172,6 +175,13 @@ extern "C" {
 #define OSSL_PKEY_PARAM_DSA_PUB_KEY  "pub"
 #define OSSL_PKEY_PARAM_DSA_PRIV_KEY "priv"
 
+/* Elliptic Curve Keys */
+#define OSSL_PKEY_PARAM_EC_PUB_KEY   "pub"
+#define OSSL_PKEY_PARAM_EC_PRIV_KEY  "priv"
+
+/* Elliptic Curve Key Parameters */
+#define OSSL_PKEY_PARAM_ECDH_COFACTOR_MODE  "ecdh-cofactor-mode"
```

Comment:
Currently this is not reachable from `EVP`, but it makes sense to have this property also for the key, not just for the `ECDH` ctx, as one of the possible values for the `ECDH` ctx is `-1`: use the default associated with the key (which in 1.1.1 can be altered using `EC_KEY_set_flags`/`EC_KEY_clear_flags`.

Reply:
I'm not entirely clear on what this cofactor mode actually does, and all I get when I search is talk of ECDH...  However, when looking at `ec_pmeth.c`, it seems there is indeed code paths that are usesd when the input key has that flag set.

Look, *nothing* stops you from adding a function `EVP_PKEY_set_cofactor_mode(EVP_PKEY *pk, int flag)` that does what you want.

Reply:
@levitte see #10838 
EC_CDH doesnt do much - it just adds an extra multiply of the private key by a scalar cofactor (and in most cases this cofactor is 1). It is however the mode that fips should use.